File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chain-prompt-model-parser.py
from common import finish

from parity_langchain import chat_anthropic, chat_openai, openai_embeddings, start  # noqa: F401


start()
from langchain_core.output_parsers import StrOutputParser
from langchain_core.prompts import ChatPromptTemplate

prompt = ChatPromptTemplate.from_messages([("system", "You are a parity assistant."), ("human", "{input}")])
chain = prompt | chat_openai(temperature=0, max_tokens=16) | StrOutputParser()
chain.invoke({"input": "Hello from the parity harness."})
finish()

File: packages/dd-trace/test/llmobs/parity/scenarios/langchain/chain-prompt-model-parser.js
'use strict'

const { finish } = require('../common')
const helpers = require('./helpers')

async function main () {
  const tracer = helpers.loadLangChainTracer()
  const { ChatPromptTemplate } = helpers.coreModule('@langchain/core/prompts')
  const { StringOutputParser } = helpers.coreModule('@langchain/core/output_parsers')
  const prompt = ChatPromptTemplate.fromMessages([
    ['system', 'You are a parity assistant.'],
    ['human', '{input}'],
  ])
  const chain = prompt.pipe(helpers.chatOpenAI({ temperature: 0, maxTokens: 16 })).pipe(new StringOutputParser())
  await chain.invoke({ input: 'Hello from the parity harness.' })
  await finish(tracer)
}

main().catch(error => {
  process.stderr.write(`${error.stack}\n`)
  process.exitCode = 1
})
